spring-jdbc/src/main/java/org/springframework/jdbc/support/KeyHolder.java

```diff
@@ -58,6 +58,21 @@
 	@Nullable
 	Number getKey() throws InvalidDataAccessApiUsageException;
 
+	/**
+	 * Retrieve the first item from the first map, assuming that there is just
+	 * one item and just one map, and that the item is an instance of provided class.
+	 * This is the typical case: a single, numeric generated key.
+	 * <p>Keys are held in a List of Maps, where each item in the list represents
+	 * the keys for each row. If there are multiple columns, then the Map will have
+	 * multiple entries as well. If this method encounters multiple entries in
+	 * either the map or the list meaning that multiple keys were returned,
+	 * then an InvalidDataAccessApiUsageException is thrown.
+	 * @return the generated key as a number
```

Comment:
So the javadoc is a little bit inconsistent with the intention of the new method. Still 'numeric key or number' is mentioned in javadoc, but we came up with the new method to overcome that limit, right?

Reply:
uh, of course you're right :) copy pasting error. Fixed.